core/src/commonMain/kotlin/com/ustadmobile/core/util/ext/ReportSeriesExt.kt

```diff
@@ -0,0 +1,163 @@
+                            whereList.add("Person.dateOfBirth <= ${dateTimeAge.dateDayStart.unixMillisLong} ")
+                        }
+
+                    }
+                }
+                ReportFilter.FIELD_PERSON_GENDER ->{
+
+                    when(filter.reportFilterCondition){
+
+                        ReportFilter.CONDITION_IS -> {
+
+                            when(filter.reportFilterDropDownValue){
+                                Person.GENDER_UNSET -> {
+                                    whereList.add("Person.gender = ${Person.GENDER_UNSET} ")
+                                }
+                                Person.GENDER_MALE -> {
+                                    whereList.add("Person.gender = ${Person.GENDER_MALE} ")
+                                }
+                                Person.GENDER_FEMALE -> {
+                                    whereList.add("Person.gender = ${Person.GENDER_FEMALE} ")
+                                }
+                                Person.GENDER_OTHER ->{
+                                    whereList.add("Person.gender = ${Person.GENDER_OTHER} ")
+                                }
+                            }
+                        }
+                        ReportFilter.CONDITION_IS_NOT ->{
+                            when(filter.reportFilterDropDownValue){
+                                Person.GENDER_UNSET -> {
+                                    whereList.add("Person.gender != ${Person.GENDER_UNSET} ")
+                                }
+                                Person.GENDER_MALE -> {
+                                    whereList.add("Person.gender != ${Person.GENDER_MALE} ")
+                                }
+                                Person.GENDER_FEMALE -> {
+                                    whereList.add("Person.gender != ${Person.GENDER_FEMALE} ")
+                                }
+                                Person.GENDER_OTHER ->{
+                                    whereList.add("Person.gender != ${Person.GENDER_OTHER} ")
+                                }
+                            }
+                        }
+
+                    }
+                }
+            }
+        }
+        if (report.toDate > 0L && report.fromDate > 0L) {
+            whereList.add("(StatementEntity.timestamp >= ? AND StatementEntity.timestamp <= ?) ")
+            paramList.add(report.fromDate)
+            paramList.add(report.toDate)
+        }
+        val whereListStr = " AND " + whereList.joinToString(" AND ")
+        sql += whereListStr
+        sqlList += whereListStr
+
+    }
+
+
+    sql += " GROUP BY xAxis "
+    if (reportSeriesSubGroup != 0) {
+        sql += " , subgroup "
+    }
+
+    sqlList += " GROUP BY StatementEntity.statementUid ORDER BY StatementEntity.timestamp DESC"
+
+
+    return QueryParts(sql, sqlList, paramList.toTypedArray())
+}
+
+private fun groupBy(value: Int): String {
```

Comment:
Strftime is not compatible with postgres. We need to be able to run all queries on postgres. The query should run on the server when the client is not fully updated.